### DISCUSSION TITLE
Is json thread-safe in writing?

### DISCUSSION THREAD

**Discussion by pam66** [nlohmann/json#3468] [Q&A] [answered]
I have this code segment

```c++
      nlohmann::json::basic_json::value_type metadata;
      ....
      long unsigned int gappixels=0;
      long unsigned int s;
      #pragma omp parallel for default(shared) private(s) reduction (+: gappixels)
      for (int ids=0;ids<nsp;ids++)
      {
        vector<coords> & cc=origPixelsCoords[ids];
        s=cc.size();
        metadata["num"][ids]=s;
        gappixels += s;
        for (long unsigned int k=0;k<s;k++)
        {
          metadata["originalCoords"][ids][k]={cc[k][0],cc[k][1],cc[k][2]};
        }
      }

```
which crashes unless I comment out the `#pragma omp` OpenMp loop parallelization directive. It's a multi-threading writing but each thread  writes only on its own `ids` location. Is this a problem? Is there any way to circumvent this? 

Thanks!

**Comment by falbrechtskirchinger** [comment, marked as answer]
`basic_json` is at most as thread-safe as the underlying types used (`std::map`, `std::vector`, ...), which, for the purpose of your question, translates to not at all. Use whatever mechanisms you'd ordinarily use to prevent data races.
